packages/data/src/utils/emitter.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Create an event emitter.
+ *
+ * @return {import("../types").WPDataEmitter} Emitter.
+ */
+export function createEmitter() {
+	let isPaused = false;
+	let isPending = false;
+	const listeners = new Set();
+	const notifyListeners = () =>
+		Array.from( listeners ).forEach( ( listener ) => listener() );
```

Comment:
I don't think we need `Array.from()` here, `Set` should support [`forEach`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/forEach) already.

Reply:
if a listener gets added as a response of a listener, it shouldn't be called (there's a unit test for that) and using the `forEach` directly breaks it, so we need another array/set when first starting to iterate.

Reply:
Cool. Might be worth to also add inline comments here as it's not really straightforward.